### PR TITLE
Update index.js

### PR DESCRIPTION
**Pull request by uzaysan**
Make baseUrl parameter function

### CODE REVIEW

**Comment by uzaysan** [comment]
This is my first PR ever to any project.

My motivation behind doing this is I have an Nginx server. Its a reverse proxy with caching system. Its acting like a CDN.
I have images and videos. I would like to seperate this two type of objects like this:

```
if(filename.endsWith(".jpeg"){
    baseUrl = image.myserver.com/files/
}
if(filename.endsWith(".mp4"){
    baseUrl = videos.myserver.com/files/;

    //Or even split more
    var c = filename.substr(0, 1);
    if (c >= '0' && c <= '9') {
        // it is a number
        baseUrl = video4.server.com
    } else {
        // it isn't
        baseUrl = videos2.server.com
    }
}
```

I hope I explained myself good. I'm not a professional. Sorry for mistakes.

**Comment by davimacedo** [comment]
Thanks for the PR. It looks good to me. Few things:
- Could you please add some test cases for the new function parameter in https://github.com/parse-community/parse-server-s3-adapter/blob/master/spec/test.spec.js?
- Could you please run `npm run lint` and fix the link problems? you can see them [here](https://travis-ci.com/github/parse-community/parse-server-s3-adapter/builds/188207151) as well.

**Comment by uzaysan** [comment]
@davimacedo when you say fix the link problems, you mean these:

```
  163:7   error  Expected space(s) after "if"                                                  keyword-spacing
  163:24  error  Missing space before opening brace                                            space-before-blocks
  166:43  error  A space is required after ','                                                 comma-spacing
  168:41  error  A space is required after ','                                                 comma-spacing
  169:9   error  Closing curly brace does not appear on the same line as the subsequent block  brace-style
  170:9   error  Expected space(s) after "else"                                                keyword-spacing
  170:13  error  Unnecessary 'else' after 'return'                                             no-else-return
  176:1   error  Trailing spaces not allowed                                                   no-trailing-spaces
  177:7   error  Block must not be padded by blank lines                                       padded-blocks

```

I believe I can fix them. 

But I dont know how to write tests. Like I said I'm not a professional. I'm a law school student. And I do programming as a hobby. 
But I will try my best.

**Comment by davimacedo** [comment]
Sorry I said link but it is lint. For the test case, you can duplicate [this][https://github.com/parse-community/parse-server-s3-adapter/blob/master/spec/test.spec.js#L318] section and add similar test cases but passing a function instead of a string. By doing this we will make sure that your addition will not only work but also nobody will break it in the future.

**Comment by uzaysan** [comment]
@davimacedo I fixed the lint problems. Now I will try to write some tests. I will edit test.spec.js file in my own branch and push it right?

**Comment by uzaysan** [comment]
@davimacedo I believe I wrote a succesfull test. You can look it [here](https://github.com/uzaysan/parse-server-s3-adapter/blob/make-baseurl-function/spec/test.spec.js#L318) And it passes every check.

**Review by davimacedo — APPROVED** [review]
Good job! LGTM!

**Comment by uzaysan** [comment]
@davimacedo Thank you. When will these changes be available on npm?

**Comment by davimacedo** [comment]
I am not sure of the exact date, but I'm working to have a new release of Parse Server soon with this.

File: spec/test.spec.js [hunk 1]
```diff
@@ -353,6 +353,49 @@ describe('S3Adapter tests', () => {
       expect(s3.getFileLocation(testConfig, 'test.png')).toEqual('https://myBucket.s3.amazonaws.com/foo/bar/test.png');
     });
   });
+  describe('getFileLocation', () => {
+    const testConfig = {
+      mount: 'http://my.server.com/parse',
```
[the remark this answers is not in the document]
Comment (TomWFox) [reply spec/test.spec.js:358]: Well I learnt something today!

**Pull request merged**